test/id_reader/schema-spec.ts

```diff
@@ -117,7 +122,10 @@ describe('id_reader Schema', () => {
                 ]
             });
 
-            expect(() => new WorkerTestHarness(job, { clients })).toThrow();
+            await expect(async () => {
+                const test = new WorkerTestHarness(job, { clients });
+                await test.initialize();
```

Comment:
You've changed the nature of this test a little bit, it's now unclear if this is meant to throw when WorkerTestHarness is intantiated or when `initialize()` is called on the resultant class.  Should the original test have had the call to `initialize` and you're just fixing that?

Reply:
Oh duh, you're also making this async, is this because of a client connection change or something?  There's a lot more of this happening lower down in the diff.

Reply:
the code that throws used to live in the constructor. However that code became async so it now has to live in the initialize function. It has to do with dynamically importing files and getting schemas to validate configs